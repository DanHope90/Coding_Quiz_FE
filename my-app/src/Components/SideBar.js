import React from "react";
import { Link } from "react-router-dom";
import "../Styles/SideBar.css";

function SideBar() {
  return (
    <div className="sidebar">
      <ul className="sidebar-links">
        <li className="sidebar-links-item">
          <Link className="sidebar-links-link" to="/register">Register</Link>
        </li>
        <li className="sidebar-links-item">
          <Link className="sidebar-links-link" to="/login">Login</Link>
        </li>
      </ul>
    </div>
  );
}
export default SideBar;
