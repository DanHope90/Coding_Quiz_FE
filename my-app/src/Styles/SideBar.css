.sidebar {
    background-color: rgb(186, 185, 185);
    display: grid;
    border-right: 2px solid black;
}

.sidebar-links {
    margin: 0;
    padding: 0;
}

.sidebar-links-item {
    list-style-type: none;
    padding: 15px 0px 15px 0px;
}

.sidebar-links-link {
    color: black;
    text-decoration: none;
    font-weight: bold;
    margin: 0;
    padding: 5px;
    display: inline-block;
}

.sidebar-links-link:active{
    color: white;
}

@media only screen and (max-width: 600px) {
    .sidebar {
        font-size: 12px;
        border-right: none;
        border-bottom: 2px solid black;
    }
    .sidebar-links{
        display: flex;
        justify-content: space-evenly;
    }
    .sidebar-links-item{
        padding: 0;
    }
}