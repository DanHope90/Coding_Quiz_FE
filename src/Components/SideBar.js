import React, { useContext } from "react";
import { Link } from "react-router-dom";
import "../Styles/SideBar.css";
import { FaRegIdCard, FaDoorOpen, FaUserCircle } from "react-icons/fa";
import { BsDoorClosed } from "react-icons/bs";
import Context from "../Utils/Context";

function SideBar() {
  const {
    isLoggedIn,
    setIsLoggedIn,
    defaultUserInfo,
    userInfo,
    setUserInfo,
  } = useContext(Context);

  return (
    <div className="sidebar">
      {(!isLoggedIn && (
      <ul className="sidebar-links">
        <li className="sidebar-links-item">
          <Link className="sidebar-links-link" to="/register">Register <FaRegIdCard /></Link>
        </li>
        <li className="sidebar-links-item">
          <Link className="sidebar-links-link" to="/">Login <FaDoorOpen /></Link>
        </li>
      </ul>
      ))}
      {(isLoggedIn && (
      <ul className="sidebar-links">
        <li className="sidebar-links-item">
          <div>{userInfo.userName}<FaUserCircle /></div>
        </li>
        <li className="sidebar-links-item">
          <button type="button" className="logout-button" onClick={() => [setIsLoggedIn(false), setUserInfo(defaultUserInfo)]}>Logout <BsDoorClosed /></button>
        </li>
      </ul>
      ))}
    </div>
  );
}
export default SideBar;
