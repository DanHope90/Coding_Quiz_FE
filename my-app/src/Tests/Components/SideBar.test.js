import SideBar from "../../Components/SideBar";
import React from "react";
import renderer from "react-test-renderer";

test("renders the SideBar correctly", () => {
    const testRenderer = renderer.create(<SideBar />);
    expect(testRenderer).toMatchSnapshot();
});
