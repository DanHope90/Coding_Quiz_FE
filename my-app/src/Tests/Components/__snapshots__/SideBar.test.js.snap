// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the SideBar correctly 1`] = `
<div
  className="sidebar"
>
  <ul
    className="sidebar-links"
  >
    <li
      className="sidebar-links-item"
    >
      <a
        className="sidebar-links-link"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </li>
    <li
      className="sidebar-links-item"
    >
      <a
        className="sidebar-links-link"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </li>
  </ul>
</div>
`;
